### ISSUE TITLE
Don't include @httpResponseCode in the response payload when there is no @httpPayload

### ISSUE THREAD

**Issue by ghostbuster91** [disneystreaming/smithy4s#1004]
`@httpResponseCode` should not be included in the response payload when there is no `@httpPayload` annotation. 

Example:
```
@http(method: "POST", uri: "/{name}", code: 200)
operation Hello {
    output: Greeting
}

structure Greeting {
    @required
    message: String

    @httpResponseCode
    @required
    code: Integer
}
```

In such case the response payload will contain the `code` field. If we change `@httpResponseCode` to `@httpHeader` the header will not be included in the response payload as expected.

AWS documentation says:
> By default, all structure members that are not bound as part of the HTTP message are serialized in a protocol-specific document sent in the body of the message (e.g., a JSON object).

**Comment by daddykotex** [comment]
That sounds like a bug.

**Comment by yisraelU** [comment]
@ghostbuster91 Can you include reproducer for this ?

**Comment by ghostbuster91** [comment]
here you go: https://github.com/ghostbuster91/demos/tree/smithy4s-1004

However, when creating that reproducer I discovered that the issue has been fixed in 0.17.9
Hence I am going to close it.
